static/js/app.js

```diff
@@ -264,36 +277,43 @@ function runCallRoulette() {
                 // constraints
                 null
             );
-        } else {
+            break;
+        default:
             throw new Error('type must be "offer" or "answer", but "' +type+ '" was given');
         }
 
-        // createAnswer or createOffer succeeded
-        function createSucceeded(desc) {
-            self._conn.setLocalDescription(
-                desc,
-                // success
-                function() {
-                    if (onSuccess) {
-                        onSuccess(self._conn.localDescription.sdp);
-                    }
-                },
-                // failure
-                function(error) {
-                    if (onFailure) {
-                        onFailure(error);
-                    }
-                }
-            );
+    }
+
+    CallRoulette.prototype.onStateChanged = function (prevState) {
```

Comment:
Now that it's a function of CallRoulette, let's make it "private" -> _onStateChanged.